base/complex.jl

```diff
@@ -631,7 +631,24 @@ function log1p(z::Complex{T}) where T
     end
 end
 
-function ^(z::Complex{T}, p::Complex{T})::Complex{T} where T<:AbstractFloat
+function exp2(z::Complex{T}) where T<:AbstractFloat
+    er = exp2(real(z))
+    theta = imag(z) * log(convert(T, 2))
+    s, c = sincos(theta)
+    Complex(er * c, er * s)
+end
+exp2(z::Complex) = exp2(float(z))
+
+function exp10(z::Complex{T}) where T<:AbstractFloat
+    er = exp10(real(z))
+    theta = imag(z) * log(convert(T, 10))
+    s, c = sincos(theta)
+    Complex(er * c, er * s)
+end
+exp10(z::Complex) = exp10(float(z))
+
+# _cpow helper function to avoid method ambiguity with ^(::Complex,::Real)
+function _cpow(z::Complex{T}, p::Union{T,Complex{T}})::Complex{T} where T<:AbstractFloat
     if p == 2 #square
```

Comment:
Note that this `p==2` special-case code dates all the way back to @jiahao in #2891, who I guess added it as a performance optimization.

Performance optimization for `z^2` is probably mainly for literal powers, though, so maybe we should move this to a `literal_pow` method?  For `Complex64` and `Complex128`, `z^2` is already inlined as `z*z`, but maybe you can save an operation or two by specializing it further.